### PR TITLE
move debug gcflags to a new var and remove one block on server build

### PR DESCRIPTION
**Pull request by trilopin**
#### Summary
Simplify `make server` command by setting a new var for gcflags based on MM_DEBUG.

it was an opportunistic improvement while checking the current tooling.

#### Ticket Link
Nope

### CODE REVIEW

**Comment by trilopin** [comment]
@hanzei kind reminder to check this tiny pr 🙏🏽

File: Makefile [hunk 1]
```diff
@@ -59,33 +65,23 @@ endif
...
-endif
+	cd server && env CGO_ENABLED=0 GOOS=linux GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-linux-amd64;
+	cd server && env CGO_ENABLED=0 GOOS=linux GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-linux-arm64;
+	cd server && env CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-darwin-amd64;
+	cd server && env CGO_ENABLED=0 GOOS=darwin GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-darwin-arm64;
+	cd server && env CGO_ENABLED=0 GOOS=windows GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) $(GO_BUILD_GCFLAGS) -trimpath -o dist/plugin-windows-amd64.exe;
 endif
 endif
 
+
```
Comment (hanzei) [inline Makefile:84]: Unrelated change

**Review by hanzei — APPROVED** [review]
The changes look good. Would you pleas also port them to https://github.com/mattermost/mattermost-plugin-starter-template?
